### DISCUSSION TITLE
custom install directory not working after upgrade

### DISCUSSION THREAD

**Discussion by jginspace** [RaspAP/raspap-webgui#866] [General]
While running version 2.5 I moved all the files under /var/www/html to /var/www/html/raspap and Raspap still worked without issues. After upgrading to version 2.6.3 none of the links down the left-hand side were working - hostapd_conf, dhcpd_conf, etc - they'd give 404 errors and my cursory debugging indicated that my clicks weren't reaching index.php.

After messing around in /etc/lighttpd I realised 50-raspap-router.conf had:
```
$HTTP["url"] =~ "^/var/www/html/raspap/(?!(dist|app|ajax|config)).*" {
      url.rewrite-once = ( "^/var/www/html/raspap/(.*?)(\?.+)?$"=>"/var/www/html/raspap/index.php/$1?$2" )
      server.error-handler-404 = "/var/www/html/raspap/index.php"
  }
```
I changed to:
```
$HTTP["url"] =~ "^/raspap/(?!(dist|app|ajax|config)).*" {
      url.rewrite-once = ( "^/raspap/(.*?)(\?.+)?$"=>"/raspap/index.php/$1?$2" )
      server.error-handler-404 = "/raspap/index.php"
  }
```
And it worked.

I also had to copy config.php from /var/www/html/raspap/config to /var/www/html/raspap/includes

I hope hope this helps someone, I found the hint via https://github.com/RaspAP/raspap-webgui/pull/813

**Comment by billz** [comment]
The quick installer will handle this for you with the `--upgrade` option, if you are upgrading from a previous version. 
It's also described in detail in the manual setup instructions.
https://docs.raspap.com/manual/#create-the-web-application

**Reply by jginspace** [reply]
I used the --upgrade option.

**Reply by billz** [reply]
The installer asks if you are installing to a directory other than the default `/var/www/html` root. If you accepted the default value, RaspAP will not be configured for your custom directory.

**Reply by jginspace** [reply]
during upgrade i entered raspap - ie /var/www/html/raspap - for the custom location ... hence 50-raspap-router.conf had:
```
$HTTP["url"] =~ "^/var/www/html/raspap/(?!(dist|app|ajax|config)).*" {
      url.rewrite-once = ( "^/var/www/html/raspap/(.*?)(\?.+)?$"=>"/var/www/html/raspap/index.php/$1?$2" )
      server.error-handler-404 = "/var/www/html/raspap/index.php"
  }
```

**Reply by billz** [reply]
I took an existing install at `/var/www/html` and manually moved it to `/var/www/html/raspap`, as in your example. 
Invoking the installer with the `--upgrade` option, I specified `/var/www/html/raspap` and the installer configured `50-raspap-router.conf` like so:

```
$HTTP["url"] =~ "^/raspap/(?!(dist|app|ajax|config)).*" {
    url.rewrite-once = ( "^/raspap/(.*?)(\?.+)?$"=>"/raspap/index.php/$1$2" )
    server.error-handler-404 = "/raspap/index.php"
}
```

After the upgrade I'm able to use the new install root at http://raspberrypi.local/raspap/ with no issues.
